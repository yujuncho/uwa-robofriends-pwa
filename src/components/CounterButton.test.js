import { shallow } from "enzyme";
import CounterButton from "./CounterButton";

describe("CounterButton", () => {
  it("expects to render", () => {
    const wrapperCounterButton = shallow(<CounterButton />);
    expect(wrapperCounterButton).toMatchSnapshot();
  });

  it("expects initial count of 1", () => {
    const wrapperCounterButton = shallow(<CounterButton />);
    const instanceCounterButton = wrapperCounterButton.instance();
    expect(instanceCounterButton.state.count).toBe(1);
    expect(instanceCounterButton.state.count).toBeLessThan(2);
  });

  it("expects button click to increment count by 1", () => {
    const wrapperCounterButton = shallow(<CounterButton />);
    const instanceCounterButton = wrapperCounterButton.instance();
    expect(instanceCounterButton.state.count).toBe(1);

    wrapperCounterButton.find("button").simulate("click");

    expect(instanceCounterButton.state.count).toBe(2);
  });

  it("expects shouldComponentUpdate to be false when the count remains the same", () => {
    const wrapperCounterButton = shallow(<CounterButton />);
    const shouldUpdate = wrapperCounterButton
      .instance()
      .shouldComponentUpdate({}, { count: 1 });
    expect(shouldUpdate).toBe(false);
  });
});
