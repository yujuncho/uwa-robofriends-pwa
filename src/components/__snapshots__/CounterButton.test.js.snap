// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton expects to render 1`] = `
<button
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
